### PR TITLE
Try this branch: Vastly improve webpack build times

### PR DESCRIPTION
**Pull request by brandonc**
## Overview

Uses various tricks to _greatly_ decrease webpack build time. On my 6 core i7 machine, initial `npm start` is about 50% faster, but successive `npm start`s are about 80% faster. Javascript hot reloading continues to work, with "watch" rebuilds taking about 4-5s

1. Turn off ExtractTextPlugin, image-webpack-plugin in local development. This was previously attempted but not actually happening

2. Parallelize our most intensive workers, babel-loader and ts-loader by pipelining files through thread-loader. I'm most interested in making sure this change doesn't break any ts syntax checking.

3. Cache the output of babel-loader (built into the package) and css loaders so that unchanged files don't need to be rebuilt. This will generated a ~16MB node_modules/.cache folder.

## Testing

Try branching off this branch for your next issue instead of master? Not ready to merge this until I make sure it doesn't mess up any workflows.

😎 

screen shot 2018-09-20 at 1 27 59 pm

### CODE REVIEW

**Comment by mperrotti** [comment]
Tried using this locally with my DCOS-41914 changes and everything seemed to work as expected

File: webpack/webpack.dev.js [hunk 1]
```diff
@@ -3,23 +3,30 @@ const merge = require("webpack-merge");
...
-  plugins: [
-    new DefinePlugin({
-      "process.env": {
-        NODE_ENV: JSON.stringify("development"),
-        LATER_COV: false
-      }
-    })
-  ]
-});
+const profile_maybe = process.env.PROFILE_WEBPACK
```
Comment (brandonc) [inline webpack/webpack.dev.js]: 🤦‍♂️ camelCase. Trying to forget ruby.

**Comment by nLight** [comment]
I'd like to hold merge this until we branch `release/1.12`

**Comment by brandonc** [comment]
@nLight I'd like to challenge you a little on the decision to hold off on this merge. I dug into the production build output from this branch and it's parent and they were exactly the same (except for source maps, which differ on every build)

To test:

1. npm install && npm run build-assets
2. mv dist/ dist-webpack
3. git checkout 607e7efdc (the branch parent)
4. npm install && npm run build-assets
5. diff --brief -r dist/ dist-webpack/

```
$ diff --brief -r dist/ dist-webpack/
Files dist/index.js.map and dist-webpack/index.js.map differ
Files dist/vendor.js.map and dist-webpack/vendor.js.map differ
```

Then we may get the benefit when working on 1.12 backports.

**Comment by brandonc** [comment]
@nLight I do agree that the webpack 4 branch can wait

**Comment by brandonc** [comment]
I rebased this on master and dismissed your review @nLight sorry!

**Comment by nLight** [comment]
I haven't experienced any significant boost. Maybe I'm doing something wrong?

**Pull request merged**